### ISSUE TITLE
This will never work on newer android devices

### ISSUE THREAD

**Issue by rfaelens** [EliteAndroidApps/WhatsApp-Key-DB-Extractor#30]
See https://android.googlesource.com/platform/frameworks/base/+/921dd75
The whole point of this workaround is to install an older version of WhatsApp to access the application storage. This is prohibited in newer versions of android.

Too bad. Please adapt the documentation accordingly.

**Comment by prateekamana** [comment]
I need my whatsapp key :( is there no other way ? Did you find a workaround ?

**Comment by rfaelens** [comment]
Use an older device with an old version of android. Even an emulator might do. 
Install WhatsApp, restore backup from Google drive, then apply the downgrade procedure. This will not be blocked in older versions of android. Just hope that WhatsApp will still install on this older version of android...

**Comment by ghost** [comment]
> I need my whatsapp key :( is there no other way ? Did you find a workaround ?

You can use WhatsDump, by MarcoG3 >> https://github.com/MarcoG3/WhatsDump.

**Comment by ghost** [comment]
> > I need my whatsapp key :( is there no other way ? Did you find a workaround ?
> 
> You can use WhatsDump, by MarcoG3 >> https://github.com/MarcoG3/WhatsDump.

You can't, its outdated and the referencing libraries and dependencies are broken.
Do this: https://community.signalusers.org/t/signal-fork-with-whatsapp-migration/21004/73?u=roberts

**Comment by Blocked** [comment]
Not true. [Your link](https://android.googlesource.com/platform/frameworks/base/+/921dd75) states 

>To downgrade a non-debuggable (i.e., release) package, uninstall thepackage (thus wiping its data), then install the older version of the package.

But The script will work with uninstalling apk, while keeping the data. See 
https://github.com/EliteAndroidApps/WhatsApp-Key-DB-Extractor/blob/a68f062df88e6189e207688fc68d873f1f87355e/WhatsAppKeyDBExtract.sh#L84-L89


Uninstalling first with data intact(``-k``) and then downgrading still works(though it's clearly a security hole). However, Recommend [this](https://github.com/YuvrajRaghuvanshiS/WhatsApp-Key-Database-Extractor)

**Comment by Blocked** [comment]
@prateekamana Try https://github.com/YuvrajRaghuvanshiS/WhatsApp-Key-Database-Extractor

**Comment by lucas75018** [comment]
@rfaelens 
That is exactly what I found out the long way when using Android 12 :-/ 

`Performing Streamed Install
adb: failed to install tmp\LegacyWhatsApp.apk: Failure [-26: Package com.whatsapp new target SDK 19 doesn't support runtime permissions but the old target SDK 31 does.]`

Have you an idea what is the latest Android version that supports such a downgrade?

Thanks and regards!
